src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -0,0 +1,3670 @@
+        hidden_states = hidden_states + residual
+        hidden_states = self.layer_norm_after(hidden_states)
+
+        return hidden_states, attention_weights
+
+
+class GroundingDINOBiMultiHeadAttention(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+
+        vision_dim = text_dim = config.d_model
+        embed_dim = config.encoder_ffn_dim // 2
+        num_heads = config.encoder_attention_heads // 2
+        dropout = config.fusion_dropout
+
+        self.embed_dim = embed_dim
+        self.num_heads = num_heads
+        self.head_dim = embed_dim // num_heads
+        self.vision_dim = vision_dim
+        self.text_dim = text_dim
+
+        if self.head_dim * self.num_heads != self.embed_dim:
+            raise ValueError(
+                f"embed_dim must be divisible by num_heads (got `embed_dim`: {self.embed_dim} and `num_heads`: {self.num_heads})."
+            )
+        self.scale = self.head_dim ** (-0.5)
+        self.dropout = dropout
+
+        self.vision_proj = nn.Linear(self.vision_dim, self.embed_dim)
+        self.text_proj = nn.Linear(self.text_dim, self.embed_dim)
+        self.values_vision_proj = nn.Linear(self.vision_dim, self.embed_dim)
+        self.values_text_proj = nn.Linear(self.text_dim, self.embed_dim)
+
+        self.out_vision_proj = nn.Linear(self.embed_dim, self.vision_dim)
+        self.out_text_proj = nn.Linear(self.embed_dim, self.text_dim)
+
+        self._reset_parameters()
+
+    def _shape(self, tensor: torch.Tensor, seq_len: int, bsz: int):
+        return tensor.view(bsz, seq_len, self.num_heads, self.head_dim).transpose(1, 2).contiguous()
+
+    def _reset_parameters(self):
+        nn.init.xavier_uniform_(self.vision_proj.weight)
+        self.vision_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.text_proj.weight)
+        self.text_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.values_vision_proj.weight)
+        self.values_vision_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.values_text_proj.weight)
+        self.values_text_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.out_vision_proj.weight)
+        self.out_vision_proj.bias.data.fill_(0)
+        nn.init.xavier_uniform_(self.out_text_proj.weight)
+        self.out_text_proj.bias.data.fill_(0)
+
+    def forward(
+        self,
+        vision_features: Tensor,
+        text_features: Tensor,
+        vision_attention_mask: Optional[Tensor] = None,
+        text_attention_mask: Optional[Tensor] = None,
+    ):
+        """_summary_
+
+        Args:
+            vision_features Tensor: bs, n_img, dim
+            text_features Tensor: bs, n_text, dim
+            vision_attention_mask (Tensor, optional): _description_. bs, n_img
+            text_attention_mask (Tensor, optional): _description_. bs, n_text
+
+        Returns:
+            _type_: _description_
```

Comment:
Docstrings should follow the format specified here: https://github.com/huggingface/transformers/tree/main/docs#testing-documentation-examples

Reply:
Fixed here https://github.com/huggingface/transformers/pull/26087/commits/a729a389db5932a68b38afccb31b28dcdfc96203